web/src/app/components/job-list-item/job-list-item.component.ts

```diff
@@ -22,10 +24,99 @@ import {GroundPinService} from 'app/services/ground-pin/ground-pin.service';
 import {DomSanitizer, SafeUrl} from '@angular/platform-browser';
 import {MatDialog} from '@angular/material/dialog';
 import {DataStoreService} from 'app/services/data-store/data-store.service';
+import {LocationOfInterestService} from 'app/services/loi/loi.service';
 import {NavigationService} from 'app/services/navigation/navigation.service';
 import {environment} from 'environments/environment';
-import {Subscription} from 'rxjs';
+import {BehaviorSubject, shareReplay, Subscription, merge, Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
 import {SurveyService} from 'app/services/survey/survey.service';
+import {SubmissionService} from 'app/services/submission/submission.service';
+import {Map as ImmutableMap} from 'immutable';
+
+/** Flat node with expandable and level information */
+export class DynamicFlatNode {
+  constructor(
+    public item: string,
+    public level = 1,
+    public expandable = false,
+  ) {}
+}
+
+export class DynamicDataSource implements DataSource<DynamicFlatNode> {
+  dataChange = new BehaviorSubject<DynamicFlatNode[]>([]);
+
+  // Data represents just the rendered data, only when lois are expanded is when
+  // submissions get added to this and to the DOM
+  get data(): DynamicFlatNode[] {
```

Comment:
We have `data` and we also have `_data` below, can we give them more specific names to differentiate? That'll help the readability a lot.

Maybe we can name them related to `submissions`? Or if you want it to be more generic, this can be `dataToRender`/`nodetoRender`?